test/e2e/framework/kcp.go

```diff
@@ -255,7 +262,14 @@ func newKcpFixture(t *testing.T, cfgs ...kcpConfig) *kcpFixture {
 		// Wait for the server to become ready
 		go func(s *kcpServer, i int) {
 			defer wg.Done()
-			err := s.Ready(!cfgs[i].RunInProcess)
+
+			ctx, cancel := context.WithCancel(context.Background())
```

Comment:
Is this right? The previous impl bound to `s.ctx` in order to stop these things before the test ended. If you don't do that you run into issues where the goroutines call e.g. `t.Error()` after the test is cancelled and then the Go test runtime panics etc

Reply:
Will look & fix if needed